### PR TITLE
Fix random_choice to handle multidim p and sizes that are not None

### LINKED ISSUE #3378: sample_prior_predictive for pm.Categorical with vector of probabilities gives only one sample
I want to sample_prior_predictive from pm.Categorical with p as a vector, but it gives me only one sample rathar than an array of samples.

For example, the code:
```python
P = [[.5,.5,0],[0,.01,.99]]
with pm.Model() as model:
    output = pm.Categorical('output',p=P,shape=(2,))
    sampling = pm.sample_prior_predictive(5)
    
sampling['output']
```
gives only:
```python
array([1, 2])
```

However, the use of only one P gives an expected result:
```python
P = [.5,.5,0]
with pm.Model() as model:
    output = pm.Categorical('output',p=P)
    sampling = pm.sample_prior_predictive(5)
    
sampling['output']
```
the output:
```python
array([0, 1, 1, 1, 1])
```

As a workaround, I can of course try to implement a for-loop in my pm.Model and loop over all Ps, but it would not look very nice. Alternativale, pm.Multinomial provides the desired result:
```python
P = [[.5,.5,0],[0,.01,.99]]
with pm.Model() as model:
    output = pm.Multinomial('output',n=1,p=P,shape=(2,3))
    sampling = pm.sample_prior_predictive(5)
    
[(np.flatnonzero(x)%3).tolist() for x in sampling['output']]
```
with output:
```python
[[1, 2], [1, 2], [0, 2], [1, 2], [1, 2]]
```
but again it looks less nicer as it would be with one-line Categorical.

**Please provide any additional information below.**

It was originally posted on [discourse](https://discourse.pymc.io/t/pm-categorical-for-matrix-of-probabilities/2766/2) and @lucianopaz confirmed that it's a bug:

> It’s a shape bug. It seems to be caused by distributions.dist_math.random_choice. I’ll try to fix it soonish, but I’m a bit swamped with other things right now.

Gist is available [here](https://gist.github.com/aakhmetz/dad012ea2ef7e9f711866fea4915eed5).

## Versions and main components

* PyMC3 Version: 3.6 (github version from 20190222)
* Theano Version: 1.0.3
* Python Version: 3.7.1
* Operating system: archlinux
* How did you install PyMC3: pip with the latest version from github

### PR DESCRIPTION
**Pull request by lucianopaz**
Fixed `dist_math.random_choice` to handle multidimensional `p` arrays and also non `None` sizes correctly. This fixes #3378.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - Added tests for mixtures of multidimensional distributions to the test suite.
 - Fixed incorrect usage of `broadcast_distribution_samples` in `DiscreteWeibull`.
 - `Mixture`'s default dtype is now determined by `theano.config.floatX`.
+- `dist_math.random_choice` now handles nd-arrays of category probabilities, and also handles non `None` sizes.
```
Comment (twiecki) [inline RELEASE-NOTES.md]: non?

Comment (lucianopaz) [reply RELEASE-NOTES.md]: Oops, I meant sizes that are not None. I'll change it in the next commits.

**Comment by lucianopaz** [comment]
If nothing unexpected happens with the tests, this should be ready to merge.

**Comment by lucianopaz** [comment]
@twiecki, what do you think about the import from `distribution` into `dist_math`? I think it's kind of ugly. Would you agree on moving the implementation of `to_tuple` into `dist_math`?

**Comment by twiecki** [comment]
Yes, that makes more sense I think.

**Review by junpenglao — APPROVED** [review]
Lgtm

**Comment by twiecki** [comment]
Thanks!

**Pull request merged**